### ISSUE TITLE
Stopped working after serverboot

### ISSUE THREAD

**Issue by GamingHQ** [flokol120/Spigot-Item-Chest-Sorter#38]
```
[18:14:46] [Server thread/ERROR]: Could not pass event InventoryCloseEvent to Item-Chest-Sorter v1.13.0
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:310) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_18_R2.event.CraftEventFactory.handleInventoryCloseEvent(CraftEventFactory.java:1376) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2228) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.network.protocol.game.PacketPlayInCloseWindow.a(SourceFile:15) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.network.protocol.game.PacketPlayInCloseWindow.a(SourceFile:6) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:31) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.server.TickTask.run(SourceFile:18) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.util.thread.IAsyncTaskHandler.d(SourceFile:157) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(SourceFile:23) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1190) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:1) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.util.thread.IAsyncTaskHandler.y(SourceFile:131) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.server.MinecraftServer.be(MinecraftServer.java:1169) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.server.MinecraftServer.y(MinecraftServer.java:1162) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.util.thread.IAsyncTaskHandler.c(SourceFile:140) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1146) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1051) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3485-Spigot-b6d12d1-2ac7fa7]
        at java.lang.Thread.run(Thread.java:833) [?:?]
```

**Comment by GamingHQ** [comment]
The following information comes with this error. 

1. When the error kicked in, the server and Minecraft in common had issues with logging into the game. It could not fetch my player information. Not sure if that's coming in handy to know.
2. Items that were placed in the chest remain seenable as a ghost objects, but still can interact with it. On rare occasions, you can even pick the items that are floating between two blocks. 
3. Items that failed to be stored and are floating do not go away when reloading the plugin. 

The only way to fix this for me was to do a hard reset on the mod. ( deleting config, plugin, reload server, add plugin, reload server ) and that worked. 

I do have a few suggestions to improve the plugin if possible. 

/ics share [friends/all] - Sharing the chest with others so they can use it as well to store items in.
/ics unshare [friends/all]- stopping with sharing listed above.
/ics remove receiver * - A wildcard to remove all receivers in one go. ( i mean, i have aproxx 700 chests a.t.m )
/ics bulk add receiver - Able to add more chests at the same time by defining the line of chests. for example, /ics bulk add receiver S10 N1. It will add all chests within 10 blocks going south of the line and including the second chest on the back to make them as large chests. 
/ics count [itemname/category/wildcard] To count specific resources or to count them all on how much you have stored.

**Comment by flokol120** [comment]
Hi @GamingHQ,

is this behavior reproducible? If yes what can I do to find this error? If not I am afraid I cannot do much. The error you posted does not mention a single class I wrote, meaning it could also be a spigot bug.

Regarding your suggestions: I like them, please open a new issue for those. If I happen to have some time to spare, I'll look into those.

**Comment by GamingHQ** [comment]
This error is not reproducible unless the login servers of Minecraft are acting up again. This error happened on the day that we had problems with the login servers. I spent 20 minutes on logging in that day as it kept thinking I was willing to boot up a demo while I'm a game pass holder. 

I'm not a huge player in the Minecraft world. i chucked myself into the deep with it. ( playing it now for a few months now ) I saw your plugin name giving the error, assuming it was yours. But, I do know that spigot has its downsides as well. 

I'm glad you like the suggestions. Sure, ill drop them in a separate treath.
